src/main/java/org/folio/consortia/config/kafka/KafkaService.java

```diff
@@ -0,0 +1,120 @@
+package org.folio.consortia.config.kafka;
+
+import java.util.Arrays;
+import java.util.List;
+
+import org.apache.commons.lang3.StringUtils;
+import org.apache.kafka.clients.admin.NewTopic;
+import org.folio.spring.FolioExecutionContext;
+import org.springframework.beans.factory.BeanFactory;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.beans.factory.config.ConfigurableBeanFactory;
+import org.springframework.core.env.Environment;
+import org.springframework.kafka.config.KafkaListenerEndpointRegistry;
+import org.springframework.kafka.config.TopicBuilder;
+import org.springframework.kafka.core.KafkaAdmin;
+import org.springframework.kafka.core.KafkaTemplate;
+import org.springframework.stereotype.Component;
+import org.springframework.util.Assert;
+
+import lombok.AllArgsConstructor;
+import lombok.Getter;
+import lombok.RequiredArgsConstructor;
+import lombok.extern.log4j.Log4j2;
+
+
+@Component
+@Log4j2
+@RequiredArgsConstructor
+public class KafkaService {
+
+  public static final String EVENT_LISTENER_ID = "mod-consortia-events-listener";
+
+  private final KafkaAdmin kafkaAdmin;
+  private final KafkaTemplate<String, Object> kafkaTemplate;
+  private final KafkaListenerEndpointRegistry kafkaListenerEndpointRegistry;
+  private final BeanFactory beanFactory;
+  private final Environment springEnvironment;
+  private final FolioExecutionContext folioExecutionContext;
+
+  @Value("${env:folio}")
+  private String environment;
+
+  @RequiredArgsConstructor
+  @AllArgsConstructor
+  @Getter
+  public enum Topic {
+    USER_CREATED("USER_CREATED"),
+    USER_DELETED("USER_DELETED"),
+    CONSORTIUM_PRIMARY_AFFILIATION_CREATED("Default", "CONSORTIUM_PRIMARY_AFFILIATION_CREATED"),
+    CONSORTIUM_PRIMARY_AFFILIATION_DELETED("Default", "CONSORTIUM_PRIMARY_AFFILIATION_DELETED");
+    private String nameSpace;
+    private final String topicName;
+  }
+
+  public void createKafkaTopics() {
+    var tenantId = folioExecutionContext.getTenantId();
+    List<NewTopic> newTopics = tenantSpecificTopics(tenantId);
+
+    log.info("Creating topics for kafka [topics: {}]", newTopics);
+    var configurableBeanFactory = (ConfigurableBeanFactory) beanFactory;
+    newTopics.forEach(newTopic -> {
+      var beanName = newTopic.name() + ".topic";
+      if (!configurableBeanFactory.containsBean(beanName)) {
+        configurableBeanFactory.registerSingleton(beanName, newTopic);
+      }
+    });
+    kafkaAdmin.initialize();
+  }
+
+  /**
+   * Restarts kafka event listeners in mod-data-export-spring application.
```

Comment:
typo here, need to change to mod-consortia

Reply:
fixed typo